lib/websocket_server.rb

```diff
@@ -15,7 +15,10 @@ def initialize
         end
 
         # Skip this loop if we can't do anything
-        next if Array(reads).empty? || writes.empty? || errors.any?
+        if Array(reads).empty? || writes.empty? || errors.any?
```

Comment:
.. oh and in that case the first part of the condition will be true.

in all other cases there should be 3 arrays